homeassistant/components/sensor/__init__.py

```diff
@@ -584,13 +584,14 @@ def state(self) -> Any:
             _LOGGER.warning(
                 (
                     "Entity %s (%s) is using native unit of measurement '%s' which "
-                    "is not a valid unit for the device class ('%s') it is using; "
+                    "is not a valid unit %s for the device class ('%s') it is using; "
```

Comment:
Grammatically this feels in the wrong place.

Maybe this works better:

```suggestion
                    "is not a valid unit for the device class ('%s') it is using; expected: %s"
```

Reply:
Had to break the line and extended the wording somewhat more.